### PR TITLE
Using order status constants in unit tests

### PR DESCRIPTION
**Pull request by wjrosa**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Base PR https://github.com/woocommerce/woocommerce/pull/51919.

This PR uses the constants introduced in https://github.com/woocommerce/woocommerce/pull/51919 in existing unit tests.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Since this changes only unit tests, code review should be enough. Check if the tests are still passing.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Uses the newly introduced order status constants in unit tests.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/php/includes/data-stores/class-wc-order-data-store-cpt-test.php
Comment (barryhughes) [inline plugins/woocommerce/tests/php/includes/data-stores/class-wc-order-data-store-cpt-test.php]: Not essential, but I personally find this a tad verbose and, actually, harder to read than the string literals that came before it. Can we import `DraftOrders` (and then update all references in the class, not just this one)?

File: plugins/woocommerce/tests/php/includes/wc-order-functions-test.php
Comment (barryhughes) [inline plugins/woocommerce/tests/php/includes/wc-order-functions-test.php]: Are these correct? Here and elsewhere across the PR, should we be referencing `OrderStatus` rather than `WC_Order`?

Comment (wjrosa) [reply plugins/woocommerce/tests/php/includes/wc-order-functions-test.php]: Oops, sorry! I forgot to update a couple of files. All of them should be good now 👍

File: plugins/woocommerce/tests/php/includes/data-stores/class-wc-order-data-store-cpt-test.php
Comment (wjrosa) [reply plugins/woocommerce/tests/php/includes/data-stores/class-wc-order-data-store-cpt-test.php]: Sure! Done ✅

**Review by barryhughes — APPROVED** [review]

**Comment by barryhughes** [comment]
Looks good, tests are passing.

**Comment by barryhughes** [comment]
@wjrosa I assume we're good to merge, but since it's stacked on top of another PR I'll leave that to you, just in case you prefer to wait.

**Comment by wjrosa** [comment]
> @wjrosa I assume we're good to merge, but since it's stacked on top of another PR I'll leave that to you, just in case you prefer to wait.

Thanks, Barry! Yeah, I think it is better to wait for the first PR to land

File: plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableDataStoreTests.php
Comment (kalessil) [inline plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableDataStoreTests.php]: Can we revert using `DraftOrders` in this PR, please? Given the core's complexity, it's better to narrow down the changes to use new constants and not going beyond.

**Review by kalessil — CHANGES_REQUESTED** [review]
Looks good! 

We should narrow down changes a bit (see my comment below) before merging the PR.

File: plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableDataStoreTests.php
Comment (wjrosa) [reply plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableDataStoreTests.php]: Absolutely! Did it on https://github.com/woocommerce/woocommerce/pull/51920/commits/ce8642684313f03ad57e5c85c674010da47498f1

**Review by kalessil — APPROVED** [review]
LGTM and CI is green!

**Pull request merged**
